Clarification for noobs.

loothunter1 opened the pull request:
About my contributions to guides.  You see, I am a new guy to all this web design stuff. I just got some basic knowledge of HTML and started to learn Ruby on Rails. So not only I had to search the web for what that SQL statement [here](http://guides.rubyonrails.org/active_record_basics.html#creating-active-record-models) actually means, but I also found out that this is not SQL statement, but **MySQL** statement which has a slightly different syntax. So, I made some clarifications for that.

Also, I think that a link to a viable MySQL tutorial/documentation for people who don't have prior knowledge of Relational Data Bases would be a good thing, but I don't know what should I better pick. So I'm waiting for suggestions in comments.

And another problem for me as a noob was [Contributing to the Rails Documentation](http://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation) itself. It says to "open a pull request", but opening a request is itself requires to make a change to source code and make a personal fork for a project. That was somewhat confusing, as I'm new to GitHub too and I didn't know that at first, I should edit the source code for guides and after that "fork" would be created automatically.

loothunter1 committed:
Update active_record_basics.md

I made a bit of clarification for people, who are not familiar with SQL (pretty much like me). However, I don't know what tutorial for MySQL is better, so I haven't inserted a link yet.

loothunter1 committed:
[ci skip] For those who new to GitHub

Added more instructions for contributing guides. Without them, it was somewhat confusing for me to find what I should actually do.

eileencodes commented on guides/source/active_record_basics.md:
````diff
@@ -152,7 +154,7 @@ CREATE TABLE products (
 ```
 
-Following the table schema above, you would be able to write code like the
+Schema above declares a table with two columns: `id` and `name`. Each row of this table represents a certain product with these two parameters. Thus, you would be able to write code like the
````
Lines in documentation should be wrapped to 80 characters. See the above where you changed `SQL` to `MySQL` and how those are wrapped? That should be done on all new documentation.

loothunter1 replied:
Done. )

eileencodes commented on guides/source/contributing_to_ruby_on_rails.md:
```diff
@@ -134,12 +134,14 @@ learn about Ruby on Rails, and the API, which serves as a reference.
 
 NOTE: To help our CI servers you should add [ci skip] to your documentation commit message to skip build on that commit. Please remember to use it for commits containing only documentation changes.
 
+NOTE2: In case you are new to GitHub, you need to register and then make an edit to Rails guide source files. Upon that a new fork will be created, which you can use in new pull request.
```
On one hand I want the guides to be as accessible as possible, on the other hand Rails shouldn't include documentation on _how_ to use GitHub. Docs on how to use GitHub is GitHub's job, so I'm leaning towards removing this.

loothunter1 replied:
The thing is that the instruction to create pull request is confusing since you can't make pull request unless you create a new branch and change the source code for documentation. Maybe it was just me being unattentive, but still, there can probably be more newcomers like me.

Ok, here is a new variant. Without "note" section, that just says that to contribute one needs to edit source files and **then** make a pull request.
[See this commit](https://github.com/rails/rails/pull/31704/commits/3bf3863b3f0a2f5ac0215db4b9bd94ddb6d458d2)

eileencodes commented on guides/source/active_record_basics.md:
```diff
@@ -142,7 +144,7 @@ end
 database. By doing this you'll also have the ability to map the columns of each
 row in that table with the attributes of the instances of your model. Suppose
-that the `products` table was created using an SQL statement like:
+that the `products` table was created using an MySQL statement like:
```
Can you change the article before `MySQL` ("an" to "a")? It's incorrect either way so we might as well improve the docs while in here.

loothunter1 replied:
Done.

eileencodes commented on guides/source/active_record_basics.md:
```diff
@@ -45,6 +45,8 @@ relationships of the objects in an application can be easily stored and
 retrieved from a database without writing SQL statements directly and with less
 overall database access code.
 
+NOTE: If you are not familiar with relational database management systems (rDBMS), but want to know more, please do this tutorial.
```
I see what you're saying about not understanding rDBMS but I don't know of a tutorial to link here so let's remove this for now.

loothunter1 replied:
I can try to find some tutorials myself, but my question is what should I pick?

SQL in general? ([Wikibook](https://upload.wikimedia.org/wikipedia/commons/6/6d/Structured_Query_Language.pdf) explains basics and that's where I learned it).

Or SQLite? As I understand Rails uses this implementation and you even have [a link to a site](https://www.sqlite.org/), but there is only a "Getting started" guide there and it's just awful for those who are new to SQL.

Or MySQL? Since it's used in Rails guides. Btw, can you answer why?

eileencodes reviewed (changes requested):
Welcome to Rails and thanks for the PR!

I requested some changes to your changes. Once you're done making them we usually only accept PR's with one commit message so we ask contributors to squash their commits. You can do that by running `git rebase -i master` to start an interactive rebase and then change word `pick` to `squash` for all the commits except the first one. Let me know if you need help!

loothunter1 committed:
Update active_record_basics.md

Fixed grammar and text wrapping as requested.

loothunter1 committed:
Update contributing_to_ruby_on_rails.md

Revised instructions.

loothunter1 committed:
Update contributing_to_ruby_on_rails.md

Typos

rafaelfranca commented on guides/source/active_record_basics.md:
```diff
@@ -142,7 +144,7 @@ end
 database. By doing this you'll also have the ability to map the columns of each
 row in that table with the attributes of the instances of your model. Suppose
-that the `products` table was created using an SQL statement like:
+that the `products` table was created using a MySQL statement like:
```
this is a SQL statement, not a MySQL statement, so let's keep it as it was.

loothunter1 replied:
No, it's not. As I've written before, **syntax is different**!
https://www.w3schools.com/sql/sql_autoincrement.asp

6temes replied:
Lol! You are explaining @rafaelfranca what is a SQL statement. xD

loothunter1 replied:
I'm sorry if that looked rude. I'm just telling what's written on the site linked above. If that's incorrect, then I apologize and ask for a better and accurate documentation (and tutorial, since I think that newcomers who don't know rDBMS should be presented with the opportunity to learn it).

kamipo replied:
Actually `auto_increment` is a MySQL extension, but regardless of whether using any extensions or not, this is still an SQL statement.

rafaelfranca replied:
It is not incorrect as MySQL statement, but a MySQL statement is a SQL statement specific for MySQL. I think using SQL statement is better in this case because we are talking about the language, so the server name should not matter

rafaelfranca replied:
Also, don't worry about being rude, like any human I can be wrong too 😃 so don't be afraid of correcting me

loothunter1 replied:
Again, my concern is not a technicality. It's just if a person, who is not familiar with SQL (like I was before this), would try to google the statement could be confused.

Btw, if we just remove AUTO_INCREMENT, would this statement be viable? Is it necessary? Since tutorial is supposed to be about Rails, the fewer nuances of SQL mentioned (especially regarding different implementations of the language), the better.

yawboakye replied:
@rafaelfranca I think I slightly appreciate @loothunter1's concern for the differentiation. For newbies who might be encountering SQL statements for the first time, this forgivable error could hamper their understanding of SQL statements down the line. Also, do note that by default Rails uses SQLite3 which doesn't have the `auto_increment` extension and so if the learner takes the pain to inspect their generated SQL they could be thrown off on an error that actually isn't. Please consider these arguments and see if we can make any changes. Thank you for making our lives easier and more productive everyday :smile:

loothunter1 committed:
Update active_record_basics.md

loothunter1 committed:
[ci skip] Update active_record_basics

Added a few links to SQL tutorials found on the net.
Also, changed MySQL to SQL (or one of its extensions) - I think that it's a good compromise.

loothunter1 committed:
[ci skip] I think now it's more clear what to do.

loothunter1 committed:
[ci skip] Fixed strings

loothunter1 commented:
Ok, I made requested changes (with modifications on my part, please review them). How do I "squash" my commits (as this was also requested)? Can I do that from Firefox browser or I have to install some soft (I'm on Windows 10)?

loothunter1 commented:
@eileencodes @rafaelfranca 
Ok, I made requested changes (with modifications on my part, please review them). How do I "squash" my commits (as this was also requested)? Can I do that from Firefox browser or I have to install some soft (I'm on Windows 10)?

rafaelfranca commented:
Thank you! I squashed using the GitHub feature but I kept the commit credits.
